core/src/main/java/com/klaviyo/coresdk/KlaviyoLifecycleCallbackListener.kt

```diff
@@ -0,0 +1,37 @@
+package com.klaviyo.coresdk
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import com.klaviyo.coresdk.networking.NetworkBatcher
+
+class KlaviyoLifecycleCallbackListener: Application.ActivityLifecycleCallbacks {
```

Comment:
This is not a necessary part of the application but outline the benefits of having a custom application lifecycle callback implemented by the parent application for us in monitoring user sessions: https://docs.google.com/document/d/10fseikYPosBoYg0daQqr6zGyfeLrn-DiIWUSdzvCmCw/edit#heading=h.tenapcbrmzfp

In this instance I am using the callbacks to empty out the batch thread before the application gets terminated. Threads will die when their parent app dies and we don't want to lose requests that may have been previously batched up.

Reply:
Why not just put this in a threaded background service? Cause services need to be registered in the manifest of the parent application. It is totally possible to make that a requirement of user setup but I think it would a question for later improvements.

Reply:
Another alternative is to use the WorkManager API to sync up our requests on the OS-level for some system-wide batching. Bonus: It can double as a push scheduler with Firebase Cloud Messaging too. This is an external dependency though so we'd like to avoid its inclusion if possible.
Detailed here: https://docs.google.com/document/d/1Hft90xwKJITFajUsLcXnhVMjfNXLXhbbhjzs9TwvUwI/edit#heading=h.tenapcbrmzfp